python/metatensor-core/tests/blocks.py

```diff
@@ -469,6 +470,28 @@ def test_to():
             assert moved.gradient("g").device.type == torch.device(device).type
 
 
+def test_values_setter():
+    block = TensorBlock(
+        values=np.array([[3.0, 4.0, 9.0]]),
+        samples=Labels.range("s", 1),
+        components=[],
+        properties=Labels.range("p", 3),
+    )
+
+    with pytest.raises(
+        AttributeError,
+        match=re.escape(
+            "Direct assignment to `values` is not allowed. "
+            "Please use block.values[:] = new_values instead."
+        ),
+    ):
+        block.values = np.array([[4, 5, 6]])
+
+    # Check that setting with slice assignment works correctly
+    block.values[:] = np.array([[4, 5, 6]])
+    assert np.allclose(block.values, np.array([[4, 5, 6]]))
```

Comment:
could you add the same test in metatensor-torch?

Reply:
I try to do it in torch but for torch I get this error that i do not know how to override:

```
In [4]: block.values=torch.tensor([[4., 5., 6.]])
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[4], line 1
----> 1 block.values=torch.tensor([[4., 5., 6.]])

RuntimeError: can't set attribute
```

Reply:
chatty fixed the problem after a bit of inputs